ai_agents/documentation_qna_agent/app.py

```
@@ -0,0 +1,151 @@
+        "load_summary": None,
+    }
+    for key, value in defaults.items():
+        if key not in st.session_state:
+            st.session_state[key] = value
+
+
+def is_valid_url(url: str) -> bool:
+    try:
+        parsed = urlparse(url.strip())
+        return parsed.scheme in {"http", "https"} and bool(parsed.netloc)
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)